pkg/pilot/cassandra/v3/pilot.go

```diff
@@ -39,9 +45,65 @@ func NewPilot(opts *PilotOptions) (*Pilot, error) {
 		nodeTool:            opts.nodeTool,
 	}
 
+	// hack to test the seedprovider works in k8s, this should be moved somewhere better
+	cfgPath := "/etc/cassandra/cassandra.yaml"
+	read, err := ioutil.ReadFile(cfgPath)
+	if err != nil {
+		return nil, err
+	}
+
+	newContents := strings.Replace(string(read),
+		"org.apache.cassandra.locator.SimpleSeedProvider",
+		"io.k8s.cassandra.KubernetesSeedProvider", -1)
+
+	newContents = strings.Replace(string(newContents),
+		"SimpleSnitch",
+		"GossipingPropertyFileSnitch", -1)
+
+	err = ioutil.WriteFile(cfgPath, []byte(newContents), 0)
+	if err != nil {
+		return nil, err
+	}
+
+	snitchSettings, err := getSnitchSettings(opts.kubeClientset)
+	if err != nil {
+		return nil, err
+	}
+
+	glog.V(4).Info("generated snitch:")
+	glog.V(4).Info(snitchSettings)
+
+	err = ioutil.WriteFile("/cassandra-rackdc.properties", []byte(snitchSettings), 0)
+	if err != nil {
+		return nil, err
+	}
+
 	return p, nil
 }
 
+func getSnitchSettings(clientset kubernetes.Interface) (string, error) {
+	nodeClientset := clientset.CoreV1().Nodes()
+
+	nodeName := os.Getenv("NODE_NAME")
+	node, err := nodeClientset.Get(nodeName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	datacenter := "navigator-datacenter-1"
+	rack := "navigator-rack-1"
```

Comment:
What if these are left blank? Is blank better than a default?

Reply:
I think they then get assigned to the cassandra defaults (`rack1` in `datacenter1`). I picked a different default so we could see if GossipingPropertyFileSnitch is enabled (if we see nodes assigned to `rack1`, we know something has gone wrong)